index.js

```diff
@@ -21,6 +21,51 @@ app.get('/api/v1/favorites', (request, response) => {
     });
 });
 
+app.get('/api/v1/favorites/:id', (request, response) => {
+  database('favorites').where('id', request.params.id).select()
+    .then((favorite) => {
+      response.status(200).json(favorite);
+    })
+    .catch((error) => {
+      response.status(500).json({ error });
+    });
+});
+
+app.post('/api/v1/favorites', (request, response) => {
+  const favorite = request.body;
+
+  var missing = [];
+
+  for (let requiredParameter of ['name', 'artist_name', 'genre', 'rating']) {
+    if (!favorite[requiredParameter]) {
+      missing.push(requiredParameter);
+    }
+  }
+
+  if (missing.length > 0) {
+    return response.status(422)
+      .send({ error: `Expected format: { name: <String>, artist_name: <String>, genre: <String>, rating: <Integer> }. You're missing "${missing.join(', ')}" properties.` });
+  }
+
+  database('favorites').insert(favorite, ['id'])
+    .then(favorite => {
+      response.status(201).json(favorite[0])
+    })
+    .catch(error => {
+      response.status(500).json({ error });
+    });
+});
+
+app.delete('/api/v1/favorites/:id', (request, response) => {
+  database('favorites').where('id', request.params.id).del()
+    .then(() => {
+      response.status(204).send();
+    })
+    .catch((error) => {
+      response.status(500).json({ error });
+    });
+});
+
```

Comment:
Why are all of these lines being added if you're only implementing delete?

Reply:
I think I just forgot to switch back to master before branching.